Mlem/Models/Trackers/Post Tracker.swift

```diff
@@ -25,53 +25,83 @@ class PostTracker: ObservableObject {
     func loadNextPage(account: SavedAccount, communityId: Int?, sort: SortingOptions?, type: FeedType) async throws {
         defer { isLoading = false }
         isLoading = true
-        
+
         let request = GetPostsRequest(
             account: account,
             communityId: communityId,
             page: page,
             sort: sort,
-            type: type
+            type: type,
+            limit: 50
         )
-        
+
         let response = try await APIClient().perform(request: request)
-        
+
         guard !response.posts.isEmpty else {
             return
         }
-        
+        Task(priority: .background) {
+            preloadImages(response.posts)
+        }
+
         add(response.posts)
         page += 1
     }
+
+    func preloadImages(_ newPosts: [APIPostView]) {
+        URLSession.shared.configuration.urlCache = AppConstants.urlCache
+        for post in newPosts {
+            if let thumbnailUrl = post.post.thumbnailUrl {
+                preloadSingleImage(url: thumbnailUrl)
+            }
+            switch post.postType {
+            case .image(let url):
+                preloadSingleImage(url: url)
+            default:
+                break
+            }
+
+        }
+    }
     
+    private func preloadSingleImage(url: URL) {
+        let task = URLSession.shared.dataTask(with: url) { data, res, err in
```

Comment:
can we convert this to use the `async` variation instead of the completion blocks? We're using `async` elsewhere throughout the project:

```swift
private func preloadSingleImage(url: URL) async {
    do {
        let request = URLRequest(url: url)
        let (data, response) = try await URLSession.shared.data(for: request)
        let cachedResponse = CachedURLResponse(response: response, data: data)
        AppConstants.urlCache.storeCachedResponse(cachedResponse, for: request)
    } catch {
        // no action required on failure...
    }
}
```
we'll need to add a few `await`'s further up and mark the `preloadImages(...)` as `async` but we should try and stick with the newer APIs from Apple 🙏